README.md

```diff
@@ -1,129 +1,148 @@
+
 ```swift
 func myContextFunction(invocation: Invocation<(String), String>) {
   let requestParam = invocation.commandArgs
   // Create HTTP request
   guard let url = URL(string: "http://myserver.net/path/\(requestParam)") else {
-    // Something went wrong...
-
-    ...
+    // Something went wrong
 
     invocation.notifyFailure(error: .encodingURLError)
+    return
   }
 
   var req = URLRequest(url: url)
   req.httpMethod = "GET"
-  req.allHTTPHeaderFields = ["Content-Type": "application/json"]
   let session = URLSession.shared
 
-  // Perform Request
+  // Perform the request
   session.dataTask(with: req) { result, res, err in
     guard let result = result else {
       // Failed getting a result from the server
 
-      ...
-
       invocation.notifyFailure(error: .networkingError)
       return
     }
 
     // Convert results to a JSON object
-    let json = JSON(data: result)
+    guard let json = (try? JSONSerialization.jsonObject(with: result, options: [])) as? [String: Any] else {
+      invocation.notifyFailure(error: .jsonDecodingError)
+      return
+    }
     // Process JSON data
 
-    ...
-
     invocation.notifySuccess()
   }.resume()
 }
 ```
 
-4. Create a CircuitBreaker instance for each context function (e.g. endpoint) you wish to circuit break:
+- Create a CircuitBreaker instance for each context function (e.g. endpoint) you wish to circuit break. The CircuitBreaker instance must specify a name for the circuit breaker, the endpoint to circuit break and the fallback function. Optional configurations include: group, timeout, resetTimeout, maxFailures, rollingWindow and bulkhead, for further details about these configuration options see the API documentation.
+
 ```swift
-let breaker = CircuitBreaker(command: myContextFunction, fallback: myFallback)
+let breaker = CircuitBreaker(name: "Circuit1", command: myContextFunction, fallback: myFallback)
 ```
-    * Must specify the fallback function and the endpoint to circuit break
-    * Optional configurations include: timeout, resetTimeout, maxFailures, rollingWindow, and bulkhead
-5. Invoke the call to the endpoint by calling the CircuitBreaker `run()` method. You should pass the corresponding arguments for the context command and fallback closures. In this sample, `myContextFunction` takes a string as its parameter while `myFallback` takes a string as its second parameter:
+
+- Invoke the call to the endpoint by calling the CircuitBreaker `run()` method. You should pass the corresponding arguments for the context command and fallback closures. In this sample, `myContextFunction` takes a string as its parameter while `myFallback` takes a string as its second parameter:
+
 ```swift
-let id: String = ...
-breaker.run(commandArgs: id, fallbackArgs: "Something went wrong.")
+let requestParam: String = "myRequestParams"
+breaker.run(commandArgs: requestParam, fallbackArgs: "Something went wrong.")
 ```
 
-###### Full Implementation:
+#### Full Implementation
 
 ```swift
 ...
```

Comment:
Did you mean to remove these three dots along with the others?

Reply:
Removed as suggested.